config/errors.yaml

```diff
@@ -1,42 +1,59 @@
 # frozen_string_literal: true
 
+# Generic errors
 5000:
   :http_status: 500
   :error_code: 5000
   :description: 'Internal server error'
-4000:
-  :http_status: 404
-  :error_code: 4000
-  :description: 'Not found'
 4001:
   :http_status: 400
   :error_code: 4001
   :description: 'Missing Parameter'
-4010:
+4002:
+  :http_status: 401
+  :error_code: 4002
+  :description: 'Cannot perform this action due to unauthenticated request'
+4003:
+  :http_status: 401
+  :error_code: 4003
+  :description: 'Invalid credentials'
+
+# User errors
+4100:
+  :http_status: 404
+  :error_code: 4100
+  :description: 'User not found'
+4102:
   :http_status: 404
-  :error_code: 4010
+  :error_code: 4102
   :description: 'User can not be deleted or updated due to user not found'
-4011:
+4103:
   :http_status: 401
-  :error_code: 4011
-  :description: 'User can not be deleted or updated due to unauthorized'
-4022:
+  :error_code: 4103
+  :description: 'User can not be deleted or updated due to unauthorized request'
+4104:
   :http_status: 422
-  :error_code: 4022
-  :description: 'Cannot create profile due to invalid paramater'
-4023:
+  :error_code: 4104
+  :description: 'Cannot create profile due to invalid parameters'
+4105:
   :http_status: 422
-  :error_code: 4023
+  :error_code: 4105
```

Comment:
I would change the error description to: `cannot update user profile to invalid parameter`